### PR TITLE
CON-4181 Specify reason for reprint and report log changes

### PR DESCRIPTION
**Pull request by ChrisJWoodcock**

### CODE REVIEW

File: src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Certificates/UpdateCertificateWithAmendReasonCommandHandlerTests/UpdateCertificateWithAmendReasonCommandHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AssessorService.Application.UnitTests.Handlers.Certificates.UpdateCertificateWithAmendReasonCommandHandlerTests
+{
+    public class UpdateCertificateWithAmendReasonCommandHandlerTests
+    {
+        [Test]
```
Comment (cofaulco) [inline src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Certificates/UpdateCertificateWithAmendReasonCommandHandlerTests/UpdateCertificateWithAmendReasonCommandHandlerTests.cs:22]: should there be a test for when GetCertificate() returns null and we throw NotFoundException?

File: src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Certificates/UpdateCertificateWithReprintReasonCommandHandlerTests/UpdateCertificateWithReprintReasonCommandHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AssessorService.Application.UnitTests.Handlers.Certificates.UpdateCertificateWithReprintReasonCommandHandlerTests
+{
+    public class UpdateCertificateWithReprintReasonCommandHandlerTests
+    {
+        [Test]
+        public async Task WhenCalled_ThenGetCertificateIsCalled_ForCertificateReference()
```
Comment (cofaulco) [inline src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Certificates/UpdateCertificateWithReprintReasonCommandHandlerTests/UpdateCertificateWithReprintReasonCommandHandlerTests.cs:23]: same as above

File: src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Certificates/UpdateCertificateWithAmendReasonCommandHandlerTests/UpdateCertificateWithAmendReasonCommandHandlerTests.cs (hunk 1 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Certificates/UpdateCertificateWithAmendReasonCommandHandlerTests/UpdateCertificateWithAmendReasonCommandHandlerTests.cs]: Added additional tests

File: src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Certificates/UpdateCertificateWithReprintReasonCommandHandlerTests/UpdateCertificateWithReprintReasonCommandHandlerTests.cs (hunk 2 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Certificates/UpdateCertificateWithReprintReasonCommandHandlerTests/UpdateCertificateWithReprintReasonCommandHandlerTests.cs]: Added additional tests

**Review by cofaulco — APPROVED** [review]

**Pull request merged**